barman/retention_policies.py

```diff
@@ -126,6 +126,25 @@ def to_json(self):
         """
         return "%s %s %s" % (self.mode, self.value, self.unit)
 
+    def propagate_retention_status_to_children(self, backup_info, report, ret_status):
+        """
+        Propagate retention status to all backups in the chain.
+        This has a side-effect. It modifies or add data to report dict.
```

Comment:
```suggestion
        Propagate retention status to all backups in the chain.

        .. note::
            This has a side-effect. It modifies or add data to *report* dict.
```

This sounds more like a note.
Besides that, sphinx recommend using `*` for quoting parameters.